### PR TITLE
add appetite multiplier setting. 0-1000% food and drink needs.

### PR DESCRIPTION
**Pull request by nsklaus**
### How to use
go to settings > options > world_defaults,  you have a new option called "hunger and thirst rate".

#### Summary
 ```SUMMARY: Content "Adds new option 'hunger and thirst rate'"```

#### Purpose of change
Making the hunger and thirst rate user modifiable. 

#### Describe the solution
This make the hunger and thirst rate user modifiable. 
Now players can choose to tweak it to their liking.
0% disable the need to feed and drink.
100% is game default: normal need for food and drink.
1000% well, makes food and drink a top priority.

The game default behavior was making my character eat and drink way too much all the time, in my opinion so I thought having an option for this would be nice.

### CODE REVIEW

File: src/options.cpp [hunk 1]
```diff
@@ -1649,7 +1649,13 @@ void options_manager::add_options_world_default()
          translate_marker( "Determines how much damage monsters can take.  A higher value makes monsters more resilient and a lower makes them more flimsy." ),
          1, 1000, 100, COPT_NO_HIDE, "%i%%"
        );
-
+    
+    mOptionsSort["world_default"]++;
+    
+    add("FOOD_WATER_NEEDS", "world_default", translate_marker( "Hunger and thirst rate" ),
+         translate_marker( "Determines the rate at which hunger and thirst grows. A lower value decreases the rate and a higher increases it." ),
```
Comment (AMurkin) [inline src/options.cpp]:
```suggestion
         translate_marker( "Determines the rate at which hunger and thirst grows.  A lower value decreases the rate and a higher increases it." ),
```

File: src/player.cpp [hunk 2]
```diff
@@ -4318,11 +4317,20 @@ void player::update_needs( int rate_multiplier )
     const bool hibernating = asleep && is_hibernating();
     const bool mouse = has_trait( trait_NO_THIRST );
     const bool mycus = has_trait( trait_M_DEPENDENT );
-    float hunger_rate = metabolic_rate();
+    
+    // set the player appetite according to the setting
+    float appetite = (get_option<int>( "FOOD_WATER_NEEDS" )/100.0);
```
Comment (AMurkin) [inline src/player.cpp]:
```suggestion
    float appetite = get_option<int>( "FOOD_WATER_NEEDS" ) / 100.0;
```

**Comment by ZhilkinSerg** [comment]
We already have distinct options for hunger and thirst:

https://github.com/CleverRaven/Cataclysm-DDA/blob/9f1a0ba4d11fe1969aacf1317430f331a0d51eff/data/json/game_balance.json#L51-L64

Why do we need another combined option?

**Comment by nsklaus** [comment]
> We already have distinct options for hunger and thirst:
> 
> [Cataclysm-DDA/data/json/game_balance.json](https://github.com/CleverRaven/Cataclysm-DDA/blob/9f1a0ba4d11fe1969aacf1317430f331a0d51eff/data/json/game_balance.json#L51-L64)

> Why do we need another combined option?


i didn't know about it. in my game, i wanted to lower the need for food and drink.
my character was needing to feed and drink way too much all the time, in my opinion.
i searched in settings > options menu, i couldn't find anything to reduce the appetite.
so i tried to create a new option for it. 

what's the proper way of tweaking the values in this json file, from the in-game UI ?

**Comment by Night-Pryanik** [comment]
> mods cannot be added or removed once a game has already started

In fact, they can.

Arbitrary multipliers in the game options ain't the best way.

**Comment by ZhilkinSerg** [comment]
I am not convinced yet we need this in options, but you better expose `PLAYER_HUNGER_RATE` and `PLAYER_THIRST_RATE` (and other game balance external options) to UI than creating new combined option for both hunger and thirst rate.

If you feel current rates are not realistic you can suggest updated values for base rates. And we also accepts mods here, so you can just suggest mod which will modify external options mentioned above.

**Comment by nsklaus** [comment]
>  but you better expose `PLAYER_HUNGER_RATE` and `PLAYER_THIRST_RATE` (and other game balance external options) to UI than creating new combined option for both hunger and thirst rate.

i will try to check if i can figure out how to do this, i mean: how to "expose external options to UI".
is there any example of this already existing where i could take hints from ?

**Comment by kevingranade** [comment]
Im not at all convinced this needs to go through the u, just open a text file and edit some values.

File: src/player.cpp [hunk 3]
```diff
@@ -1,5 +1,4 @@
 #include "player.h"
-
```
[the remark this answers is not in the document]
Comment (nsklaus) [reply src/player.cpp]: no idea, you sure i did this ?
if it's indeed me then it must be a mistake.

[the remark this answers is not in the document]
Comment (BevapDin) [reply src/player.cpp]: Github says it's been introduced by ccd7b560e1cc6450708861e9b4a1f95133ffcd86

**Comment by kevingranade** [comment]
>  why use UI at all then ?

For things it is common and meaningful for the player to adjust. Fiddling with low level balance variables are not somethingcwe expect every player to do.

> "for fine tune go edit the file game_balance.json"

Yes something like this would be reasonable.

Probably we could populate config/balance.json and/or save/world/balance.json with values set to "default", and if the player wants they can override them there.

**Pull request closed without merging**